address.go

```diff
@@ -119,6 +172,61 @@ func DecodeAddress(addr string, defaultNet *chaincfg.Params) (Address, error) {
 	}
 }
 
+// decodeSegWitAddress parses a bech32 encoded segwit address string and
+// returns the witness version and witness program byte representation.
+func decodeSegWitAddress(hrp, address string) (byte, []byte, error) {
+	// We MUST verify that the human-readable part is "bc" for mainnet and "tb"
+	// for testnet.
+	hrp = strings.ToLower(hrp)
+	if hrp != "bc" && hrp != "tb" {
```

Comment:
This'll trivially break in the case that additional prefixes are added. For example, with this, one isn't able to re-use this library for Litecoin or any other coins that've adopted `bech32`. 

To fix this, the `chaincfg.Params` should be passed in and the `hrp` extracted compared against the defined `Bech32HRPSegwit` for that network.